subsys/net/ip/net_pkt.c

```diff
@@ -974,6 +974,10 @@ static size_t pkt_buffer_length(struct net_pkt *pkt,
 		if (net_if_l2(net_pkt_iface(pkt)) ==
 		    &NET_L2_GET_NAME(ETHERNET)) {
 			max_len += sizeof(struct net_eth_hdr);
+#if defined(CONFIG_NET_DSA)
```

Comment:
I would avoid such ifdef here.
Instead in include/ethernet.h create a macro like NET_ETH_MAX_HDR_SIZE  which is by default sizeof(struct net_eth_hdr)
if DSA is enabled: add +1 (have a macro for the DSA tag size as well).

Finally here, replace sizeof(struct net_eth_hdr) by NET_ETH_MAX_HDR_SIZE

Reply:
Ok. Thanks for the idea. I've rewritten the code.